packages/api-client/src/client.ts

```diff
@@ -0,0 +1,176 @@
+//
+// Copyright © 2024 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import {
+  type Account,
+  type Class,
+  type Client,
+  type Data,
+  type Doc,
+  type DocumentQuery,
+  type FindOptions,
+  type FindResult,
+  type Hierarchy,
+  type ModelDb,
+  type Ref,
+  type Space,
+  type WithLookup,
+  type TxResult,
+  DocumentUpdate,
+  TxOperations,
+  concatLink
+} from '@hcengineering/core'
+import client, { clientId } from '@hcengineering/client'
+import { addLocation, getResource, setMetadata } from '@hcengineering/platform'
+
+import { login, selectWorkspace } from './account'
+import { type APIClient, type ConnectOptions, type ConnectSocketOptions } from './types'
+
+interface ServerConfig {
+  ACCOUNTS_URL: string
+}
+
+/** @public */
+export async function connect (url: string, options: ConnectOptions): Promise<APIClient> {
+  const config = await loadServerConfig(url)
+
+  const { endpoint, token } = await getWorkspaceToken(url, options, config)
+  return await createClient(endpoint, token, options)
+}
+
+/** @public */
+export async function createClient (endpoint: string, token: string, options: ConnectSocketOptions): Promise<APIClient> {
+  addLocation(clientId, () => import(/* webpackChunkName: "client" */ '@hcengineering/client-resources'))
+
+  setMetadata(client.metadata.ClientSocketFactory, options.socketFactory)
```

Comment:
let's move all this 3 as parameters to client factory.

Reply:
done